.github/workflows/publish-release-reusable.yml

```diff
@@ -0,0 +1,118 @@
+    - name: Setup dotnet
+      uses: actions/setup-dotnet@v1.9.0
+      with:
+        dotnet-version: ${{env.dotnet-version}}
+    - name: C++ for ubuntu 18.04
+      if: matrix.os == 'ubuntu-18.04'
+      run: |
+        sudo apt-get install -y build-essential
+    - name: Choose the right C++ for ubuntu 18.04
+      if: matrix.os == 'ubuntu-18.04'
+      run: |
+        sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 60
+    - uses: actions/setup-python@v1
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - uses: actions/setup-node@v1
+    - run: npm install bignumber.js
+    - name: Install latex pandoc
+      run: |
+        brew install pandoc
+        brew install --cask basictex
+        eval "$(/usr/libexec/path_helper)"
+        sudo tlmgr update --self
+        sudo tlmgr install framed tcolorbox environ trimspaces unicode-math
+        pandoc -v
+        which latex || echo NOT FOUND latex
+        which xelatex || echo NOT FOUND xelatex
+        sudo gem install rouge
+    # First we build the ZIPs (which do not include the refman)
+    - name: Package release files (release)
+      if: ${{ !inputs.prerelease }}
+      run: |
+        python dafny/Scripts/package.py ${{ inputs.name }} --skip_manual=true --github_secret=${{ secrets.GITHUB_TOKEN }}
+    - name: Package release files (prerelease)
+      if: ${{ inputs.prerelease }}
+      run: |
+        python dafny/Scripts/package.py ${{ inputs.name }} --trial=true --skip_manual=true --github_secret=${{ secrets.GITHUB_TOKEN }}
+    # Then we build the reference manual separately, because that depends on evaluating the path_helper script,
+    # and that perturbs the path enough that `python` above would resolve to an older built-in Python version.
+    # Additionally, since the refman build scripts expect to find Dafny in its usual Binaries/ folder (not in
+    # a platform-specific directory), we build Dafny once here.
+    - name: Build Dafny
+      run: dotnet build dafny/Source/Dafny.sln
+    - name: Build reference manual
+      run: |
+        eval "$(/usr/libexec/path_helper)"
+        make -C dafny/docs/DafnyRef
+
+    - name: Create GitHub release
+      uses: softprops/action-gh-release@v1
+      with:
+        name: Dafny ${{ inputs.name }}
+        tag_name: ${{ inputs.tag_name }}
+        body: ${{ inputs.release_notes }}
+        draft: ${{ inputs.draft }}
+        prerelease: ${{ inputs.prerelease }}
+        files: |
+          dafny/Package/dafny-${{ inputs.name }}*
+          dafny/docs/DafnyRef/DafnyRef.pdf
+        fail_on_unmatched_files: true
+
+    - name: Create release NuGet package (for uploading)
+      if: ${{ !inputs.prerelease }}
+      run: dotnet pack --no-build dafny/Source/Dafny.sln
+    - name: Create prerelease NuGet package (for uploading)
+      if: ${{ inputs.prerelease }}
+      run: dotnet pack --version-suffix ${{ inputs.name }} --no-build dafny/Source/Dafny.sln
```

Comment:
It wasn't obvious to me that Nuget automatically considers any version with a suffix as a prerelease, so probably worth a comment here. Very nice that it's this simple though!